metadata/metadata_test.go

```diff
@@ -207,4 +295,203 @@ var _ = Describe("Metadata", func() {
+				ObjectMeta: v1.ObjectMeta{
+					Labels: map[string]string{"foo": "bar", "baz": "qux"},
+				},
+			}
+			podDest := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Labels: map[string]string{"quux": "corge"},
+				},
+			}
+			CopyLabelsWithPrefixReplacement(podSource, podDest, "fo", "ba")
+			CopyLabelsByPrefix(podSource, podDest, "fo")
+			Expect(podDest.Labels).To(gstruct.MatchAllKeys(gstruct.Keys{
+				"bao":  Equal("bar"),
+				"foo":  Equal("bar"),
+				"quux": Equal("corge"),
+			}))
+		})
+
+		It("should copy all the labels matching the given prefix when Labels in Dest is empty", func() {
+			podSource := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Labels: map[string]string{"foo": "bar", "baz": "qux"},
+				},
+			}
+			podDest := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{},
+			}
+			err := CopyLabelsWithPrefixReplacement(podSource, podDest, "fo", "ba")
+			err = CopyLabelsByPrefix(podSource, podDest, "fo")
+			Expect(err).To(BeNil())
+			Expect(podDest.Labels).To(gstruct.MatchAllKeys(gstruct.Keys{
+				"bao": Equal("bar"),
+				"foo": Equal("bar"),
+			}))
+		})
+
+		It("should copy all the labels matching the given prefix when Labels in Dest is empty", func() {
+			podSource := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Labels: map[string]string{"foo": "bar", "baz": "qux"},
+				},
+			}
+			podDest := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{},
+			}
+			CopyLabelsByPrefix(podSource, podDest, "fo")
+			Expect(podDest.Labels).To(gstruct.MatchAllKeys(gstruct.Keys{
+				"foo": Equal("bar"),
+			}))
+		})
+
+		It("should copy nothing to the Dest when Labels in Source object is nil ", func() {
+			podSource := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{},
+			}
+			podDest := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Labels: map[string]string{"quux": "corge"},
+				},
+			}
+			err := CopyLabelsWithPrefixReplacement(podSource, podDest, "fo", "ba")
+			Expect(err).To(BeNil())
+			err = CopyLabelsByPrefix(podSource, podDest, "fo")
+			Expect(err).To(BeNil())
+			Expect(podDest.Labels).To(gstruct.MatchAllKeys(gstruct.Keys{
+				"quux": Equal("corge"),
+			}))
+		})
+
+		It("should return error when Source is nil", func() {
+			podDest := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Labels: map[string]string{"quux": "corge"},
+				},
+			}
+			err := CopyLabelsWithPrefixReplacement(nil, podDest, "fo", "ba")
+			Expect(err).NotTo(BeNil())
+			err = CopyLabelsByPrefix(nil, podDest, "fo")
+			Expect(err).NotTo(BeNil())
+			Expect(podDest.Labels).To(gstruct.MatchAllKeys(gstruct.Keys{
+				"quux": Equal("corge"),
+			}))
+		})
+
+		It("should return error when Dest is nil", func() {
+			podSource := &corev1.Pod{
+				ObjectMeta: v1.ObjectMeta{
+					Labels: map[string]string{"quux": "corge"},
+				},
+			}
+			err := CopyLabelsWithPrefixReplacement(podSource, nil, "fo", "ba")
+			Expect(err).NotTo(BeNil())
+			err = CopyLabelsByPrefix(podSource, nil, "fo")
+			Expect(err).NotTo(BeNil())
+		})
+	})
```

Comment:
Some of my comments apply to multiple lines. Leaving this comment here to apply to the whole file:
* Split tests and don't use the same When clausule for multiple functions.
* Use Succeed() whenever is possible.
* Wrap functions returning a single value with Expect if the output won't be used in other tests.
* Add a new line to separate resources creation from tests.
* Re order code. When clausules should follow the same order in which the actual functions are defined (alphabetical order).

Reply:
Thanks for your comments. It helps a lot! I have updated the code according to them. Would you please review it one more time? Thanks!